src/v/container/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 find_package(Roaring REQUIRED)
 v_cc_library(
   NAME container
-  DEPS Roaring::roaring
```

Comment:
```
FAILED: lib/libv_v_io.so
: && /vectorized/llvm/bin/clang++ -fPIC -fPIC -march=westmere -fuse-ld=lld -Wno-unused-lambda-capture -Wno-unused-command-line-argument -stdlib=libc++ -fsanitize=undefined -fsanitize=address -fstack-protector-strong -fno-plt -fstack-clash-protection -O0 -g -gdwarf-4 -gsplit-dwarf  -Wl,--build-id=sha1 -L/vectorized/lib -L/vectorized/lib64 -stdlib=libc++ -fsanitize=undefined -fsanitize=address -Wl,-z,relro,-z,now -shared -Wl,-soname,libv_v_io.so -o lib/libv_v_io.so src/v/io/CMakeFiles/v_io.dir/persistence.cc.o src/v/io/CMakeFiles/v_io.dir/page.cc.o src/v/io/CMakeFiles/v_io.dir/page_set.cc.o src/v/io/CMakeFiles/v_io.dir/clang-tidy-helper.cc.o src/v/io/CMakeFiles/v_io.dir/logger.cc.o src/v/io/CMakeFiles/v_io.dir/io_queue.cc.o src/v/io/CMakeFiles/v_io.dir/scheduler.cc.o  -Wl,-rpath,/vectorized/lib64  /vectorized/lib64/libseastar.so  /vectorized/lib64/libboost_program_options.so.1.82.0  /vectorized/lib64/libboost_thread.so.1.82.0  /vectorized/lib64/libboost_atomic.so.1.82.0  /vectorized/lib64/libboost_chrono.so.1.82.0  /vectorized/lib64/libboost_date_time.so.1.82.0  /vectorized/lib64/libboost_container.so.1.82.0  /vectorized/lib64/libboost_exception.a  /vectorized/lib64/libcares.so  /vectorized/lib64/libcryptopp.so  /vectorized/lib64/libfmtd.so.8.1.1  -Wl,--as-needed  /vectorized/lib64/liblz4.so  -fsanitize=address  -fsanitize=undefined  -fno-sanitize=vptr  /vectorized/lib64/libabsl_cord.so.2308.0.0  /vectorized/lib64/libabsl_cordz_info.so.2308.0.0  /vectorized/lib64/libabsl_cord_internal.so.2308.0.0  /vectorized/lib64/libabsl_cordz_functions.so.2308.0.0  /vectorized/lib64/libabsl_cordz_handle.so.2308.0.0  /vectorized/lib64/libabsl_crc_cord_state.so.2308.0.0  /vectorized/lib64/libabsl_crc32c.so.2308.0.0  /vectorized/lib64/libabsl_crc_internal.so.2308.0.0  /vectorized/lib64/libabsl_crc_cpu_detect.so.2308.0.0  /vectorized/lib64/libabsl_str_format_internal.so.2308.0.0  /vectorized/lib64/libabsl_raw_hash_set.so.2308.0.0  /vectorized/lib64/libabsl_hash.so.2308.0.0  /vectorized/lib64/libabsl_bad_optional_access.so.2308.0.0  /vectorized/lib64/libabsl_city.so.2308.0.0  /vectorized/lib64/libabsl_bad_variant_access.so.2308.0.0  /vectorized/lib64/libabsl_low_level_hash.so.2308.0.0  /vectorized/lib64/libabsl_hashtablez_sampler.so.2308.0.0  /vectorized/lib64/libabsl_exponential_biased.so.2308.0.0  /vectorized/lib64/libabsl_synchronization.so.2308.0.0  /vectorized/lib64/libabsl_stacktrace.so.2308.0.0  /vectorized/lib64/libabsl_graphcycles_internal.so.2308.0.0  /vectorized/lib64/libabsl_kernel_timeout_internal.so.2308.0.0  /vectorized/lib64/libabsl_time.so.2308.0.0  /vectorized/lib64/libabsl_civil_time.so.2308.0.0  /vectorized/lib64/libabsl_time_zone.so.2308.0.0  /vectorized/lib64/libabsl_symbolize.so.2308.0.0  /vectorized/lib64/libabsl_strings.so.2308.0.0  /vectorized/lib64/libabsl_string_view.so.2308.0.0  /vectorized/lib64/libabsl_throw_delegate.so.2308.0.0  /vectorized/lib64/libabsl_strings_internal.so.2308.0.0  /vectorized/lib64/libabsl_int128.so.2308.0.0  /vectorized/lib64/libabsl_debugging_internal.so.2308.0.0  /vectorized/lib64/libabsl_malloc_internal.so.2308.0.0  /vectorized/lib64/libabsl_demangle_internal.so.2308.0.0  /vectorized/lib64/libabsl_base.so.2308.0.0  /vectorized/lib64/libabsl_raw_logging_internal.so.2308.0.0  /vectorized/lib64/libabsl_log_severity.so.2308.0.0  /vectorized/lib64/libabsl_spinlock_wait.so.2308.0.0  /vectorized/lib64/libroaring.so  -lunordered_dense && :
ld.lld: error: unable to find library -lunordered_dense
```

internal build is failing because `cluster` doesn't depend on `v::container`.

Reply:
Why does it try to link against unordered_dense though, it's a header-only library.

Reply:
Oh because you put it in deps, it will try and link against it

Reply:
Maybe trying adding `unordered_dense::unordered_dense` instead?

Reply:
Oh that will only work for the OSS build. There is no cmake target for this in the superbuild so cmake tries to link against it, I think thats why the :: alias pattern emerged. I am sort of out of my depth here, I am not a CMake wizard

Reply:
Thoughts: either need a conditional add library for oss or make a dummy target in vtools

Reply:
Thanks for those pointers, so I also have no idea what I am doing here but with: https://github.com/redpanda-data/vtools/compare/stephan/find-unordered-dense?expand=1 

and 

```
diff --git a/src/v/container/CMakeLists.txt b/src/v/container/CMakeLists.txt
index 7b260ee289..c524beffdf 100644
--- a/src/v/container/CMakeLists.txt
+++ b/src/v/container/CMakeLists.txt
@@ -1,7 +1,11 @@
 find_package(Roaring REQUIRED)
+find_package(unordered_dense REQUIRED)
+
 v_cc_library(
   NAME container
-  DEPS Roaring::roaring unordered_dense
+  DEPS
+    Roaring::roaring
+    unordered_dense::unordered_dense
 )

 add_subdirectory(tests)
```

it seems to work. 

@dotnwat is this the right way to go?

We really need to get away from the dual build.

Reply:
Everything looks ok.

When it's not a header-only library we can usually get away with a little alias hack in the open-source build like this https://github.com/redpanda-data/redpanda/blob/dev/cmake/dependencies.cmake#L135-L137.

But given it's header-only I think [FindUnorderedDense.cmake](https://github.com/redpanda-data/vtools/compare/stephan/find-unordered-dense?expand=1#diff-c270fc1d95547bc979eb41db03cc46f894fb9c10b2df03e29b7a627f70929b15) is the most reasonable thing. And that looks fine.

> We really need to get away from the dual build.

100%

Reply:
Thanks Noah, here it is: https://github.com/redpanda-data/vtools/pull/2594